### PR TITLE
Break down map methods and events into sections for docs

### PR DESCRIPTION
**Pull request by mourner**
This PR annotates `Map` methods and events in a way that divides the enormous list into named sections for use in the API reference. This makes it much easier for new users to get a handle of what the API is capable of if they don't know specifically what method name to look for. This is similar to how the [Leaflet docs](https://leafletjs.com/reference-1.7.1.html) are structured, and has been very helpful there.

However, this might be one of the worst docs-related hacks we've ever had, because I'm trying to work around `documentation.js` not supporting section tags (see also https://github.com/documentationjs/documentation/issues/128), but in the end it seems to work and produces this kind of categorization in the docs (while not interfering with search):

image

There are multiple gotchas with this approach:

- I'm adding sections inline in the source with a custom JSDoc tag `/** @section Title */`. This produces `documentation lint` warnings about unrecognized tag, which we need to silence somehow to make the build green. Additionally, I had to reorder a few methods in the source to lie under the right section.
- To be able to show section headers alongside methods/events they relate to, I had to disable alpha sorting introduced in https://github.com/mapbox/mapbox-gl-js-docs/pull/272 so that the docs follow natural flow of the source code. This might have consequences in unexpected places not related to the `Map` class. However, I'm not too worried — related methods usually come together in the source, making for a natural flow, and if there's a problem, we can reorder the code manually.
- The categorization itself (section names, etc.) should be revised — I just made an initial pass of what made sense to me but it might not be as friendly to new users.
- For some unclear reason, `resize` and `remove` events jump from its section to the top of the list, even though they're not documented any differently from the rest. I suspect this might be a `documentation.js` sorting bug which needs to be fixed upstream, but we can land without waiting for it because it's not a big deal.
- Perhaps it would make more sense for `Camera` methods to come before `Map` methods in the sorting, but I'm not sure how to control this and it also might require an upstream change. Not a deal breaker as well.

Since I'm going on vacation for the following two weeks, feel free to make any necessary further changes to be able to land this without consulting me.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] manually test the built docs
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

File: src/ui/events.js [hunk 1]
```diff
@@ -770,6 +775,11 @@ export type MapEvent =
      */
     | 'touchcancel'
 
+    /** @section Movement
```
Comment (asheemmamoowala) [inline src/ui/events.js:778]: `Movement` is a weird category. Can it be folded into a single `Querying`,  `Map State` or `Camera` category?

**Comment by HeyStenson** [comment]
@katydecorah and/or @danswick can you review this before we merge to confirm that these changes will play nice with our existing docs setup?

File: src/ui/events.js (hunk 1 above)
Comment (mourner) [reply src/ui/events.js:778]: `Camera` sounds good to me!

**Comment by katydecorah** [comment]
> can you review this before we merge to confirm that these changes will play nice with our existing docs setup?

@HeyStenson I can confirm that it does. Vlad created https://github.com/mapbox/mapbox-gl-js-docs/pull/649 as proof that this work of adding sections will work as expected.

I'm going to add this topic to docs scrum today. I think it would be good to have more docs team members review the categories.

**Comment by arindam1993** [comment]
the only major change I had to make to get this green was to wrap sections in `{}` because that was the easiest way to get `jsdocs/require-complete-sentences` to ignore those comments. I think we can parse out the `{}` on the rendering side?

**Comment by katydecorah** [comment]
>  I think we can parse out the {} on the rendering side?

@arindam1993 Confirmed, I updated mapbox-gl-js-docs PR to remove the `{` `}`.

**Review by danswick — APPROVED** [review]
❤️ This is excellent. The built site looks great and the sections all seem like a perfectly reasonable place to start. While this is a bit of a workaround as mentioned, it seems manageable enough and we can always make a push to contribute to documentation.js later if the workarounds become unmanageable!

**Pull request merged**